src/main/java/it/grid/storm/info/SpaceInfoManager.java

```diff
@@ -316,28 +319,39 @@ private void fakeSAtoAnalyze(List<String> absPaths) {
 		log.info("Background DU tasks size: {}", bDUTasks.getTasks().size());
 	}
 
+	private List<TSpaceToken> retrieveSTtoInitializeWithQuota() {
+		log.debug("Retrieving the list of SpaceToken with quota to initialize");
+		List<TSpaceToken> result = new ArrayList<TSpaceToken>();
+		List<StorageSpaceData> toInitialize = retrieveSSDtoInitializeWithQuota();
+		for (StorageSpaceData ssd : toInitialize) {
+			if (ssd.getSpaceToken() == null) {
+				log.debug("Null spacetoken found for {}", ssd);
+				continue;
+			}
+			result.add(ssd.getSpaceToken());
+			log.debug("Added {}", ssd.getSpaceToken());
+		}
+		log.debug("Returned {} elements", result.size());
+		return result;
+	}
+	
 	private void foundSAtoAnalyze() {
 
 		List<StorageSpaceData> toAnalyze = spaceCatalog
 			.getStorageSpaceNotInitialized();
+		List<TSpaceToken> quotaEnabledST = retrieveSTtoInitializeWithQuota();

```

Comment:
Use collection filtering with a comparator instead of building another collection with only the tokens.